d4s/src/main/scala/d4s/codecs/D4SAttributeEncoder.scala

```diff
@@ -16,44 +15,54 @@ trait D4SAttributeEncoder[T] {
   def postprocessAttributeEncoder(f: AttributeValue => AttributeValue): D4SAttributeEncoder[T] = item => f(encode(item))
 }
 
-object D4SAttributeEncoder {
-  @inline def apply[A](implicit ev: D4SAttributeEncoder[A]): ev.type = ev
-
-  def derived[T]: D4SAttributeEncoder[T] = macro Magnolia.gen[T]
-
-  def encode[T: D4SAttributeEncoder](item: T): AttributeValue                                 = D4SAttributeEncoder[T].encode(item)
-  def encodeField[T: D4SAttributeEncoder](name: String, item: T): Map[String, AttributeValue] = Map(name -> D4SAttributeEncoder[T].encode(item))
-
-  def traitEncoder[A](caseMap: A => (String, D4SAttributeEncoder[? <: A])): D4SAttributeEncoder[A] = {
-    item =>
-      val typeNameEncoder = caseMap(item)
-      if (typeNameEncoder._2.isInstanceOf[CaseObjectEncoder[?]]) {
-        typeNameEncoder._2.asInstanceOf[D4SAttributeEncoder[A]].encode(item)
-      } else {
-        AttributeValue.builder().m(Map(typeNameEncoder._1 -> typeNameEncoder._2.asInstanceOf[D4SAttributeEncoder[A]].encode(item)).asJava).build()
-      }
-  }
-
-  /** Magnolia instances */
-  private[D4SAttributeEncoder] type Typeclass[T] = D4SAttributeEncoder[T]
-
-  def combine[T](ctx: ReadOnlyCaseClass[D4SAttributeEncoder, T]): D4SAttributeEncoder[T] = {
+private[codecs] abstract class GenericD4SAttributeEncoder(dropNulls: Boolean) {
+  def combineImpl[T](ctx: ReadOnlyCaseClass[D4SAttributeEncoder, T]): D4SAttributeEncoder[T] = {
```

Comment:
This is only used once in `combine` i guess separate `Impl` method is not required since `dropNulls` is a class parameter, not method parameter anyway now